backend/cmd/headlamp.go

```diff
@@ -760,6 +807,58 @@ func (c *HeadlampConfig) handleClusterRequests(router *mux.Router) {
 	handleClusterAPI(c, router)
 }
 
+// Handles stateless cluster requests if kubeconfig is set and dynamic clusters are enabled.
+// It returns context key which is used to store the context in the cache.
+func (c *HeadlampConfig) handleStatelessReq(r *http.Request, kubeConfig string) (string, error) {
```

Comment:
I feel like this should all be in package rather than in the headlamp.go? @yolossn what do you think?

Reply:
have made sure new logic that is introduced is part of existing packages wherever possible. I don't think this stateless part alone needs a separate package.

Reply:
Agreed. Since the new logic about stateless cluster does not need specific implementation, creating a new package is not necessary.

Reply:
If not a package, then I still feel the code should be separated into its own file.

Keeping it in headlamp.go mixes it in with a lot of things and it makes it hard to know what goes with what.

Reply:
Done.